### PR TITLE
Initial OSIRIS prototype

### LINKED ISSUE #26: Create the OSIRIS Script
Refer back to emails to ensure that we create it in line with requirements.

### PR DESCRIPTION
**Pull request by Pasarus**
Closes #26 

## Description
Create the initial OSIRIS autoreduction prototype script

### CODE REVIEW

File: OSIRIS/reduce.py [hunk 1]
```diff
@@ -1 +1,86 @@
...
+unit_x = "DeltaE"
+fold_multiple_frames = False
+
+# Generated
+sum_runs = len(input_runs) > 1
+
+input_file_paths = ""
+for input_run in input_runs:
+    input_file_paths += ", " + generate_input_path_for_run(input_run, cycle)
+input_file_paths = input_file_paths[2:]
```
Comment (keiranjprice101) [inline OSIRIS/reduce.py:37]: Can you add a comment here explaining this.

File: OSIRIS/reduce.py [hunk 2]
```diff
@@ -1 +1,86 @@
...
+print(input_file_paths)
+
+output_workspaces = []
+
+# Perform the reduction
+if spectroscopy_reduction:
+    output_workspace_prefix = instrument
+    for input_run in input_runs:
+        output_workspace_prefix += input_run + ", "
+    output_workspace_prefix = output_workspace_prefix[:-2] + f"_{analyser}_{reflection}_Reduced"
```
Comment (keiranjprice101) [inline OSIRIS/reduce.py:47]: Similar to above the final 2 char index feels magic number-ish

File: OSIRIS/reduce.py (hunk 1 above)
Comment (Pasarus) [reply OSIRIS/reduce.py:37]:
```suggestion
input_file_paths = input_file_paths[2:]  # Slice out the excess ", "
```

File: OSIRIS/reduce.py (hunk 2 above)
Comment (Pasarus) [reply OSIRIS/reduce.py:47]:
```suggestion
    output_workspace_prefix = output_workspace_prefix[:-2] + f"_{analyser}_{reflection}_Reduced"  # Slice out the excess ", " and finalize prefix
```

**Review by keiranjprice101 — APPROVED** [review]

**Pull request merged**